plugin/src/main/java/org/opensearch/ml/utils/MLNodeUtils.java

```diff
@@ -89,6 +90,41 @@ public static void validateSchema(String schemaString, String instanceString) th
         }
     }
 
+    /**
+     * This method processes the input JSON string and replaces the string values of the parameters with JSON objects if the string is a valid JSON.
+     * @param inputJson The input JSON string
+     * @return The processed JSON string
+     */
+    public static String processRemoteInferenceInputDataSetParametersValue(String inputJson) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode rootNode = mapper.readTree(inputJson);
+
+        if (rootNode.has("parameters") && rootNode.get("parameters").isObject()) {
+            ObjectNode parametersNode = (ObjectNode) rootNode.get("parameters");
+
+            parametersNode.fields().forEachRemaining(entry -> {
+                String key = entry.getKey();
+                System.out.println(key);
+                JsonNode value = entry.getValue();
+
+                if (value.isTextual()) {
```

Comment:
Actually no, this is because if you construct a invalid Json, it will cause parsing exception and thrown earlier in rest/transport layer. If you construct a valid Json, if parameters is either exist, or empty, or with wrong type (array, string, etc.) it will skip. If it is the correct type, then it must have entries.

Reply:
Add more UTs to address your concern.